From 934c200bf34c7c8c696e27b6fcb68a5bdc37470e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Fri, 28 Nov 2025 02:51:47 +0200
Subject: [PATCH 08/14] Prevent constants from being dropped from archives due
 to inlining

on macOS with clang 13, these constants are missing when linking
otherwise.
---
 src/interpreter/cling/lib/Utils/AST.cpp   | 2 +-
 src/interpreter/cling/lib/Utils/Paths.cpp | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/interpreter/cling/lib/Utils/AST.cpp b/src/interpreter/cling/lib/Utils/AST.cpp
index fe257d90..147d3898 100644
--- a/src/interpreter/cling/lib/Utils/AST.cpp
+++ b/src/interpreter/cling/lib/Utils/AST.cpp
@@ -170,7 +170,7 @@ namespace utils {
     return result;
   }
 
-  const char* const Synthesize::UniquePrefix = "__cling_Un1Qu3";
+  const char* const Synthesize::UniquePrefix __attribute__((used)) = "__cling_Un1Qu3";
 
   IntegerLiteral* Synthesize::IntegerLiteralExpr(ASTContext& C, uintptr_t Ptr) {
     const llvm::APInt Addr(8 * sizeof(void*), Ptr);
diff --git a/src/interpreter/cling/lib/Utils/Paths.cpp b/src/interpreter/cling/lib/Utils/Paths.cpp
index b92665f5..d96e6aa6 100644
--- a/src/interpreter/cling/lib/Utils/Paths.cpp
+++ b/src/interpreter/cling/lib/Utils/Paths.cpp
@@ -20,9 +20,9 @@ namespace utils {
 
 namespace platform {
 #if defined(LLVM_ON_UNIX)
-  const char* const kEnvDelim = ":";
+  const char* const kEnvDelim __attribute__((used)) = ":";
 #elif defined(_WIN32)
-  const char* const kEnvDelim = ";";
+  const char* const kEnvDelim __attribute__((used)) = ";";
 #else
   #error "Unknown platform (environmental delimiter)"
 #endif
-- 
2.52.0

